src/fairseq2/gang.py

```diff
@@ -707,6 +707,113 @@ def fake_gangs(device: Device) -> Gangs:
     return Gangs(gang, gang, gang)
 
 
+def setup_2D_mesh_gangs(
```

Comment:
My preference would be to make this an internal function (i.e. `_setup_2d_mesh_gangs()`). We already expose the actual functionality in `setup_parallel_gangs()` and `to_fsdp()` that are higher-level and abstract away the details of the mesh creation.

Reply:
I implemented this but added an extra "setup_" function in `gang.py` to keep it coherent with `setup_parallel_gangs()`.